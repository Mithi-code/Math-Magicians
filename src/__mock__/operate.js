export default {
  one: null,
  two: null,
};
