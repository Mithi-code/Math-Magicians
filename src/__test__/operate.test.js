import operate from '../logic/operate';

describe('Operate', () => {
  const num1 = 4;
  const num2 = 2;
  const addition = operate(num1, num2, '+');
  test('The test check for  Addition of two number', () => {
    expect(addition).toBe('6');
    expect(addition).toMatchSnapshot();
  });
  const subtraction = operate(num1, num2, '-');
  test('The test check for subtraction of two number', () => {
    expect(subtraction).toBe('2');
    expect(subtraction).toMatchSnapshot();
  });
  const division = operate(num1, num2, 'รท');
  test('The test check for division of two number', () => {
    expect(division).toBe('2');
    expect(division).toMatchSnapshot();
  });
  const multiplication = operate(num1, num2, 'x');
  test('The test check for multiplication of two number', () => {
    expect(multiplication).toBe('8');
    expect(multiplication).toMatchSnapshot();
  });
  const noDivisibility = operate(num1, '0', 'รท');
  test('The  test that check for no divisibility by denominator zero', () => {
    expect(noDivisibility).toBe("Can't divide by 0.");
    expect(noDivisibility).toMatchSnapshot();
  });
  const modulus = operate(num1, num2, '%');
  test('The  test that checks for the modulus of a number', () => {
    expect(modulus).toBe('0');
    expect(modulus).toMatchSnapshot();
  });
});