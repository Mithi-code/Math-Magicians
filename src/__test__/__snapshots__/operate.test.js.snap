// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operate  The test check for  Addition of two number 1`] = `"6"`;

exports[`Operate  The test check for division of two number 1`] = `"2"`;

exports[`Operate The  test that check for no divisibility by denominator zero 1`] = `"Can't divide by 0."`;

exports[`Operate The  test that checks for the modulus of a number 1`] = `"0"`;

exports[`Operate The test check for multiplication of two number 1`] = `"8"`;

exports[`Operate The test check for subtraction of two number 1`] = `"2"`;
